### PR TITLE
f-http@v0.6.0 - Collect and Push Dependency Timings

### PR DESCRIPTION
**Pull request by tommcclean**
This PR updates f-http to continue the development of the dependency timings MVP.

- The main change is to accept an instance of f-statistics when initialising a client, doing so means that interceptors record response timings and publish a statistic with the details.
- Several files were changed to objects with constructors because not doing so means that configuration can bleed when more than one instance is initiaised.

### CODE REVIEW

File: packages/services/f-http/src/createClient.js [hunk 1]
```diff
@@ -74,7 +74,10 @@ export default (options = {}) => {
         }
     });
 
-    interceptors.addTimingInterceptor(_axiosInstance, _configuration.isDevelopment);
+    if (statsClient) {
+        // Only add interceptors when capturing statistics
+        interceptors.captureResponseStatistics(_axiosInstance, statsClient);
```
Comment (tommcclean) [inline packages/services/f-http/src/createClient.js:79]: See if I can do a type check

File: packages/services/f-http/src/createClient.js [hunk 2]
```diff
@@ -59,7 +59,7 @@ const deleteResource = async (resource, headers = {}) => _sendRequest(httpVerbs.
  * @param {object} options - Any options to override - refer to documentation for options
  * @return {object} - Returns an object with restful request methods
  */
-export default (options = {}) => {
+export default (options = {}, statsClient = null) => {
```
Comment (tommcclean) [inline packages/services/f-http/src/createClient.js:62]: What happened to the CTOR

File: packages/services/f-http/src/createClient.js [hunk 3]
```diff
@@ -3,90 +3,98 @@ import defaultOptions from './defaultOptions';
...
- * Patch a resource
- * @param {string} resource - The resource to patch (URL)
- * @param {object} body - The request body, contents of the resource
- * @param {object} headers - Any additional request headers you want to provide
- * @return {object} - Returns data from response
- */
-const patchResource = async (resource, body, headers = {}) => _sendRequestWithBody(httpVerbs.PATCH, resource, body, headers);
+        if (statsClient) {
+            // Only add interceptors when capturing statistics
+            interceptors.captureResponseStatistics(this.axiosInstance, statsClient);
```
Comment (tommcclean) [inline packages/services/f-http/src/createClient.js:31]: See if I can change this to a type check

File: packages/services/f-http/src/createClient.js [hunk 4]
```diff
@@ -3,90 +3,98 @@ import defaultOptions from './defaultOptions';
...
- */
-const deleteResource = async (resource, headers = {}) => _sendRequest(httpVerbs.DELETE, resource, headers);
+    /**
+     * Get a resource
+     * @param {string} resource - The resource to get (URL)
+     * @param {object} headers - Any additional request headers you want to provide
+     * @return {object} - Returns data from response
+     */
+    async get (resource, headers = {}) {
+        return this.sendRequest(httpVerbs.GET, resource, headers);
```
Comment (tommcclean) [inline packages/services/f-http/src/createClient.js:47]: Does this need to be awaited

File: packages/services/f-http/src/interceptors.js [hunk 5]
```diff
@@ -22,16 +16,19 @@ const addTimingInterceptor = ({ interceptors }, isDevelopment) => {
...
-            // TODO : Output to stats engine
-        }
+        const payload = {
+            'cs-method': res.config.method.toUpperCase(),
+            'cs-uri-stem': `${res.config.baseURL}${res.config.url}`,
+            'sc-status': res.status,
+            'time-taken': res.responseTimeMs
+        };
+
+        statisticsClient.publish(`${payload['cs-method']} request to ${payload['cs-uri-stem']} took ${payload['time-taken']}ms`, payload);
```
Comment (tommcclean) [inline packages/services/f-http/src/interceptors.js:26]: Be more defensive

File: packages/services/f-http/src/createClient.js [hunk 6]
```diff
@@ -3,90 +3,98 @@ import defaultOptions from './defaultOptions';
...
- * @param {string} resource - The resource to get (URL)
- * @param {object} headers - Any additional request headers you want to provide
- * @return {object} - Returns data from response
+ * Create a httpClient
+ * @param {object} options - Any options to override - refer to documentation for options
+ * @return {object} - Returns an object with restful request methods
  */
-const getResource = async (resource, headers = {}) => _sendRequest(httpVerbs.GET, resource, headers);
+export default class StatClient {
+    constructor (options = {}, statsClient = null) {
```
Comment (tommcclean) [inline packages/services/f-http/src/createClient.js:14]: ℹ️ - This enables more than one instance to be created

**Review by kevinrodrigues — APPROVED** [review]

File: packages/services/f-http/CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,14 @@
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+v0.6.0
+------------------------------
+*July 26, 2021*
+
+### Changed
+- Accept an instance of f-statistics when initialising a client, doing so means that interceptors record response timings and publish a statistic with the details.
+- Several files were changed to objects with constructors because not doing so means that configuration can bleed when more than one instance is initiaised.
```
Comment (kevinrodrigues) [inline packages/services/f-http/CHANGELOG.md:12]: https://thumbs.gfycat.com/LonelySlushyGalapagostortoise-mobile.mp4

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**